### DISCUSSION TITLE
What network architecture to use to setup Security Onion

### DISCUSSION THREAD

**Discussion by nicholaswkc34** [Security-Onion-Solutions/securityonion#3407] [Unsupported Versions] [answered]
Dear All, 
I'm plann to set up security onion at my home network but i don't know which network architecture to follow. Can anyone please tell me what network architecture to use if i have current network setup as below: 

Modem - pfsense Box - Bridge WIFI Router - Computer

Please help. A billion thanks for your help.

**Comment by nicholaswkc34** [comment]
How to mirror all traffic from pfsense to SO?

**Comment by weslambert** [comment, marked as answer]
Here is an example (can ignore the Suricata inline component):

https://dfirmadness.com/building-a-siem-at-home/

**Reply by vanh20** [reply]
Hello peterwkc85,

Weslambert's link is comprehensive. If you follow the instructions, it gives you a better way to secure your network and have a homelab. However, if you just want a simpler and less costly solution, you can just add a five port mirroring switch (with one port capable of mirroring function) and get another network connection for your computer. Configure your port mirroring switch in such a way that ports 1 to 4 port gets mirrored to port 5. Connect port 1 to where your wifi router is previously connected to pfsense. Then connect your SecurityOnion machine's sniffing port at port 5.

You then have to convert/configure your wifi router into a wifi access point. There are links on the internet on how to do this. What you want to happen is this: your pfsense machine becomes the DHCP/DNS server/provider. Remember, your wifi router cannot have DHCP enabled. To make things simple, we can't have more than one DHCP server on the same broadcast domain. It can't have firewall enabled also unless it is allowed by your wifi router's firmware in "Wifi Access Point" mode. You will not use the WAN port of this device either (again unless allowed). There is a way for some routers (if you are inclined to experiment), to feed the WAN port back to one of its LAN port in this configuration, but it may or may not work for your router and it will require you to configure your wifi lan port address manually to a different subnet offered by your pFsense' DHCP. In my case, I needed to do this in order to maintain a logging function built-in on my router, but this is off topic already.

Then you need to connect one of your wifi router's lan ports to one of the ports (possibly port 2) of your port mirroring switch. You also connect your computer to one of the unused ports (let's say port 3) on your port mirroring switch. So you still have one unused port that you can use for other purposes.

Also, consider how your wifi accesss point work. SO will not be able to monitor traffic that flows between your wifi clients and your access point's 4 port switch since they operate in bridge mode. If you have a wifi router that has Access Point mode built-in and still have a working firewall (prevents all devices from talking to each other, i.e. "connect to internet only" function), then you're in such luck! You would want this feature. :)

Next, you need another network card or a usb to lan device on your computer, and you need to connect that to your SecurityOnion's management port. Configure it manually in such a way that this network node and your SO management port are in the same subnet, different from what pFsense' DHCP dishes out. You can have another computer to handle this function instead of doing it on one computer as per your configuration. But if you are just trying things out and not so much particular of security, this will do. Ideally, you'd want all your management and security devices on a secured subnet or VLAN separate from any possible compromised clients that may connect on your network.

I hope this works for you. Sorry that I cannot go into the details. Just find your way on the internet on each device you need to configure.
